### PR TITLE
Revert "Remove 'authenticated with sso' param from MHV URL helper (#1…

### PR DESCRIPTION
**Pull request by ncksllvn**
…5562)"

This reverts commit 0198171528dd1b81bcb367dacfbd83fae5f9f44d.

## Description
Per the request of the MHV team, we need to revert the work we did with sending all users to the "eauth" domain. 

https://github.com/department-of-veterans-affairs/va.gov-team/issues/17318

## Testing done
Did not test since this is just a direct revert. MHV will confirm that it's been reverted okay.

## Screenshots
N/A

## Acceptance criteria
- [ ] "eauth" domain is only used for SSO users

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by micahchiang — APPROVED** [review]

File: src/applications/static-pages/health-care-manage-benefits/get-medical-records-page/components/AuthContent/index.js [hunk 1]
```diff
@@ -29,7 +33,7 @@ const AuthContent = ({ cernerFacilities, otherFacilities }) => (
       cernerFacilities={cernerFacilities}
       otherFacilities={otherFacilities}
       linksHeaderText="Get your medical records from:"
-      myHealtheVetLink={mhvUrl('download-my-data')}
+      myHealtheVetLink={mhvUrl(authenticatedWithSSOe, 'download-my-data')}
```
Comment (kelsonic) [inline src/applications/static-pages/health-care-manage-benefits/get-medical-records-page/components/AuthContent/index.js:36]: hahaha I feel like we've seen this before... 😂

**Review by erikphansen — APPROVED** [review]

**Review by kelsonic — APPROVED** [review]
I never would have thought it'd be so difficult to create the correct link 😂 Way to keep up with this!! 🔥 I'm completely lost lol

**Comment by ncksllvn** [comment]
So that this makes the daily I also merged [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/15616#pullrequestreview-566553623) into this branch because the former fixes a bug on master (which would block me from merging)

**Review by craigwheeler — APPROVED** [review]

**Review by theodur — APPROVED** [review]

**Review by bosawt — CHANGES_REQUESTED** [review]
Is there a commit that wasn't intended to be part of this merge? It looks like one of the commits doesn't belong. Lemme know, otherwise I'll 👍

**Comment by ncksllvn** [comment]
Hello @bosawt , yes there is, as I mentioned in my comment above, https://github.com/department-of-veterans-affairs/vets-website/pull/15615#issuecomment-758851202

**Review by bosawt — APPROVED** [review]
Ah, sounds good!

**Pull request merged**